ballerina/tests/08_get_subfields.bal

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) 2023, WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/test;
+
+final readonly & string[] missionSubFieldNames = ["id", "designation", "startDate", "endDate"];
+final readonly & string[] astronautSubFieldNames = ["id", "name", "missions"];
+
+@test:Config {
+    groups: ["getSubfields"],
+    dataProvider: fieldObjectProvider1
+}
+function testGetSubfields1(Field 'field) {
+    Field[]? subFields = 'field.getSubfields();
+    if subFields is () {
+        test:assertFail(msg = "subfields of 'field is null");
+    }
+    test:assertEquals(subFields.length(), missionSubFieldNames.length());
+
+    foreach Field subField in subFields {
+        test:assertFalse(missionSubFieldNames.indexOf(subField.getName()) is (), msg = "subfield name not found");
+        test:assertTrue(subField.getSubfields() is (), msg = "subfields of subfield is not null");
+        test:assertTrue(subField.getPath() == ["mission", subField.getName()], msg = "path of subfield is not correct");
+    }
+}
+
+function fieldObjectProvider1() returns (Field[][]) {
+    return [
+        [field_object1],
+        [field_object4]
```

Comment:
What's the reason behind the names here?
- Names should use camelCase.
- Why do we have 1 and 4 here, 2 and 5 in another, etc.?